ras_party/controllers/account_controller.py

```diff
@@ -95,26 +93,62 @@ def post_respondent(party, tran, session):
         'status': RespondentStatus.CREATED
     }
 
-    #  Create the enrolment respondent-business-survey associations
-    respondent = Respondent(**translated_party)
-    br = BusinessRespondent(business=business, respondent=respondent)
-    pending_enrolment = PendingEnrolment(case_id=case_id,
-                                         respondent=respondent,
-                                         business_id=business_id,
-                                         survey_id=survey_id)
-    Enrolment(business_respondent=br,
-              survey_id=survey_id,
-              status=EnrolmentStatus.PENDING)
-    session.add(respondent)
-    session.add(pending_enrolment)
-    session.commit()
+    respondent = _add_enrolment_and_auth(business, business_id, case_id, party, session, survey_id,
+                                         translated_party)
+
+    disable_iac(party['enrolmentCode'], case_id)  # calls raise for status
+
     _send_email_verification(respondent.party_uuid, party['emailAddress'])
 
-    register_user(party, tran)
-    disable_iac(party['enrolmentCode'], case_id)
     return respondent.to_respondent_dict()
 
 
+def _add_enrolment_and_auth(business, business_id, case_id, party, session, survey_id, translated_party):
+    """Create and persist new party entities and attempt to register with auth service.
+    Auth fails lead to party entities being rolled back.
+    The Contxet manager commits to session, If db fails after that and before main commit then db state is unknown
+    """
+    try:
+        with session.begin_nested():
+
+            # Use a sub transaction to store party data
+            # Context manager will manage commits/rollback
+
+            # Create the enrolment respondent-business-survey associations
+
+            respondent = Respondent(**translated_party)
+            br = BusinessRespondent(business=business, respondent=respondent)
+            pending_enrolment = PendingEnrolment(case_id=case_id,
+                                                 respondent=respondent,
+                                                 business_id=business_id,
+                                                 survey_id=survey_id)
+            Enrolment(business_respondent=br,
+                      survey_id=survey_id,
+                      status=EnrolmentStatus.PENDING)
+
+            session.add(respondent)
+            session.add(pending_enrolment)
+
+    except Exception as exception:
```

Comment:
Can we have this work for all the specific exceptions instead of just the generic exception?